source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -1876,6 +1886,24 @@ void COpenGLDriver::SAuxContext::flushState(GL_STATE_BITS stateBits)
             glDepthMask(nextState.rasterParams.depthWriteEnable);
             UPDATE_STATE(rasterParams.depthWriteEnable);
         }
+        if (STATE_NEQ(rasterParams.multisampleEnable)) {
```

Comment:
rasterParams is our pseudo-Vulkan struct right?

Reply:
nah, rasterParams is member of SOpenGLState

Reply:
Can you give me a list of things like `multisampleEnable` that are in your `SOpenGLState` and not in our pseudo-Vulkan struct?

Reply:
everything from multisampleEnable down to primitiveRestartEnable, just those few things

Reply:
why do you track `primitiveRestartEnable` and `depthRange` ?

Reply:
`primitiveRestartEnable` because it depends on pipeline
`depthRange` - not sure, most likely it was in your state tracker or used somewhere.. idk

Reply:
>` primitiveRestartEnable` because it depends on pipeline

How does it depend on the pipeline?

Reply:
SPrimitiveAssemblyParams::primitiveRestartEnable and SPrimitiveAssemblyParams is obviously member of IRenderpassIndependentPipeline, corresponds to `VkPipelineInputAssemblyStateCreateInfo::primitiveRestartEnable` which is member of `VkGraphicsPipelineCreateInfo `

Reply:
ok, cool then I can resolve.